modules/develop/pages/data-transforms/how-transforms-work.adoc

```diff
@@ -0,0 +1,41 @@
+= How Data Transforms Work
+:description: Learn how Redpanda data transforms work.
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
```

Comment:
Took this from the `run-transform` doc because I think it's also important to provide examples here. I was looking for concrete examples while reading this doc.